### PR TITLE
Device and gpu protocols updated; public IJsonSerializable

### PR DESCRIPTION
**Pull request by semuserable**
Related to https://github.com/getsentry/sentry-unity/issues/49, check latest comment (which starts with 'Proposed changes').

I made `IJsonSerializable` public, because we'll have a new context on Unity side. Although, I'm not sure about `summary` description.

I also plan to update the docs, but after this part will be merged.

~I plan to update the tests too. Still WIP.~

### CODE REVIEW

**Comment by semuserable** [comment]
Added the tests. Please, review @Tyrrrz @bruno-garcia as I'd like to start the integration on Unity side for these protocols.

**Comment by lucas-zimerman** [comment]
Please consider adding a changelog entry for the next release.

Instructions and example for changelog
Please add an entry to `CHANGELOG.md` to the "Unreleased" section. Make sure the entry includes this PR's number.

Example:
`Device and gpu protocols updated; public IJsonSerializable([#1063](https://github.com/getsentry/sentry-dotnet/pull/1063))`
If none of the above apply, you can opt out of this check by adding `#skip-changelog` to the PR description.

File: src/Sentry/Protocol/Device.cs [hunk 1]
```diff
@@ -383,6 +527,17 @@ public static Device FromJson(JsonElement json)
             var screenDensity = json.GetPropertyOrNull("screen_density")?.GetSingle();
             var screenDpi = json.GetPropertyOrNull("screen_dpi")?.GetInt32();
             var bootTime = json.GetPropertyOrNull("boot_time")?.GetDateTimeOffset();
+            var processorCount = json.GetPropertyOrNull("processor_count")?.GetInt32();
+            var cpuDescription = json.GetPropertyOrNull("cpu_description")?.GetString();
+            var processorFrequency = json.GetPropertyOrNull("processor_frequency")?.GetInt32();
+            var deviceType = json.GetPropertyOrNull("device_type")?.GetString();
+            var batteryStatus = json.GetPropertyOrNull("battery_status")?.GetString();
+            var deviceUniqueIdentifier = json.GetPropertyOrNull("battery_status")?.GetString();
```
Comment (lucas-zimerman) [inline src/Sentry/Protocol/Device.cs:535]:
```suggestion
            var deviceUniqueIdentifier = json.GetPropertyOrNull("device_unique_identifier")?.GetString();
```

**Comment by lucas-zimerman** [comment]
Perhaps using const names for the field names, for example:

```
internal const string ProcessorCountKey = "processor_count";
...
if (ProcessorCount is {} processorCount)           
 {                
writer.WriteNumber(ProcessorCountKey , processorCount);           
 }
...
var processorCount = json.GetPropertyOrNull(ProcessorCountKey)?.GetInt32();

```

**Comment by semuserable** [comment]
> Perhaps using const names for the field names, for example:
> 
> ```
> internal const string ProcessorCountKey = "processor_count";
> ...
> if (ProcessorCount is {} processorCount)           
>  {                
> writer.WriteNumber(ProcessorCountKey , processorCount);           
>  }
> ...
> var processorCount = json.GetPropertyOrNull(ProcessorCountKey)?.GetInt32();
> ```

I haven't seen such usage from within the project, so used directly.

File: src/Sentry/Protocol/Device.cs (hunk 1 above)
Comment (semuserable) [reply src/Sentry/Protocol/Device.cs]: Ah.. Good catch!

File: test/Sentry.Tests/Protocol/Context/GpuTests.cs [hunk 2]
```diff
@@ -66,6 +85,11 @@ public void Clone_CopyValues()
             Assert.Equal(sut.MultiThreadedRendering, clone.MultiThreadedRendering);
             Assert.Equal(sut.Version, clone.Version);
             Assert.Equal(sut.NpotSupport, clone.NpotSupport);
+            Assert.Equal(sut.MaxTextureSize, clone.MaxTextureSize);
+            Assert.Equal(sut.GraphicsShaderLevel, clone.GraphicsShaderLevel);
+            Assert.Equal(sut.SupportsDrawCallInstancing, clone.SupportsDrawCallInstancing);
+            Assert.Equal(sut.SupportsRayTracing, clone.SupportsRayTracing);
+            Assert.Equal(sut.SupportsComputeShaders, clone.SupportsComputeShaders);
```
Comment (bruno-garcia) [inline test/Sentry.Tests/Protocol/Context/GpuTests.cs:92]:
```suggestion
            Assert.Equal(sut.SupportsComputeShaders, clone.SupportsComputeShaders);
            Assert.Equal(sut.SupportsGeometryShaders, clone.SupportsGeometryShaders);
```

File: src/Sentry/IJsonSerializable.cs [hunk 3]
```diff
@@ -9,7 +9,10 @@
 
 namespace Sentry
 {
-    internal interface IJsonSerializable
+    /// <summary>
+    /// Sentry JsonSerializable.
+    /// </summary>
+    public interface IJsonSerializable
```
Comment (bruno-garcia) [inline src/Sentry/IJsonSerializable.cs:15]: @Tyrrrz are we good with this decision?

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**